desk/sur/hooks.hoon

```diff
@@ -23,62 +25,67 @@
       state=vase
       config=(map nest config)
   ==
-::  $hooks: collection of hooks, the order they should be run in, and
-::          any delayed hooks that need to be run
+::  $hooks: collection of hooks, the order they should be run in, hooks
+::          running on a schedule, and any hooks waiting to run
 ++  hooks
-  $:  hooks=(map id hook)
-      order=(map nest (list id))
-      crons=(map id (map origin cron))
-      delayed=(map delay-id [=origin delayed-hook])
+  $:  hooks=(map id-hook hook)
+      order=(map nest (list id-hook))
+      crons=(map id-hook cron)
+      waiting=(map id-wait [=origin waiting-hook])
   ==
 +$  origin  $@(~ nest)
-+$  delay-id  id
-+$  schedule  [next=@da repeat=@dr]
-+$  cron
-  $:  hook=id
++$  cron  (map origin job)
++$  job
+  $:  =id-hook
       =schedule
       =config
   ==
-::  $delayed-hook: metadata for when a delayed hook fires from the timer
-+$  delayed-hook
-  $:  id=delay-id
-      hook=id
++$  schedule  [next=@da repeat=@dr]
++$  id-wait  @uv
+::  $waiting-hook: metadata for when a waiting hook fires from the timer
++$  waiting-hook
+  $:  id=id-wait
+      hook=id-hook
       data=vase
       fires-at=time
   ==
-::
+::  $config: configuration data for a hook instance
 +$  config  (map @t *)
+::
+::  $action: what we can do with a hook
 +$  action
```

Comment:
Perhaps we should already follow the convention here and elsewhere of `a-*` for actions and `r-*` for responses?
```suggestion
+$  a-hook
```

Reply:
This can wait, but eventually when hooks are integrated we will have to do the renaming.

Reply:
yeah I will leave this for later when we move hooks into channel actions